pkg/machine/env/dir.go

```diff
@@ -158,3 +170,19 @@ func WithPodmanPrefix(name string) string {
 	}
 	return name
 }
+
+func getDataDir() string {
```

Comment:
Might be better named `getDataSubdir` or `getDataDirSuffix` to emphasize that it's not at the same level as `GetConfigHome`, the latter defines the prefix for the dir, and the former defines its suffix.
When I started reading the PR, I was confused by `data, err := homedir.GetDataHome()` and `machineDataDir := getDataDir()`, I thought both calls were redundant.